packages/block-library/src/read-more/index.php

```diff
@@ -19,15 +19,22 @@ function render_block_core_read_more( $attributes, $content, $block ) {
 	}
 
 	$post_ID            = $block->context['postId'];
+	$post_title         = get_the_title( $post_ID );
+	/* translators: %s is either the post title or post ID to describe the link for screen readers. */
+	$screen_reader_text = sprintf(
+		__( ': %s' ),
+		$post_title !== '' ? $post_title : __( 'post ' ) . $post_ID
+	);
 	$justify_class_name = empty( $attributes['justifyContent'] ) ? '' : "is-justified-{$attributes['justifyContent']}";
 	$wrapper_attributes = get_block_wrapper_attributes( array( 'class' => $justify_class_name ) );
 	$more_text          = ! empty( $attributes['content'] ) ? wp_kses_post( $attributes['content'] ) : __( 'Read more' );
 	return sprintf(
-		'<a %1s href="%2s" target="%3s">%4s</a>',
+		'<a %1s href="%2s" target="%3s">%4s<span class="screen-reader-text">%5s</span></a>',
```

Comment:
There needs to be a space between the visible link text and the hidden text, otherwise the words will be pasted together. e.g. `<a %1s href="%2s" target="%3s">%4s<span class="screen-reader-text"> %5s</span></a>`

See the space inside the `screen-reader-text` span.

Reply:
To make "grammatical" sense in cases where the user has customized the "read more" text, I went with the following format:

`Read More: Post Title`

See https://github.com/WordPress/gutenberg/pull/45490/files#diff-5fed9d866f33e17d5bf917f312bcae02af3c8bd244586ab161ed60889c960af6R25

Does that make sense?

Reply:
Great. Also good for internationalization.